centrallix-lib/include/clusters.h

```diff
@@ -0,0 +1,137 @@
+/* Centrallix Core                                                      */
+/*                                                                      */
+/* Copyright (C) 1998-2012 LightSys Technology Services, Inc.           */
+/*                                                                      */
+/* This program is free software; you can redistribute it and/or modify */
+/* it under the terms of the GNU General Public License as published by */
+/* the Free Software Foundation; either version 2 of the License, or    */
+/* (at your option) any later version.                                  */
+/*                                                                      */
+/* This program is distributed in the hope that it will be useful,      */
+/* but WITHOUT ANY WARRANTY; without even the implied warranty of       */
+/* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the        */
+/* GNU General Public License for more details.                         */
+/*                                                                      */
+/* You should have received a copy of the GNU General Public License    */
+/* along with this program; if not, write to the Free Software          */
+/* Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA             */
+/* 02111-1307  USA                                                      */
+/*                                                                      */
+/* A copy of the GNU General Public License has been included in this   */
+/* distribution in the file "COPYING".                                  */
+/*                                                                      */
+/* Module:      lib_cluster.c, lib_cluster.h                            */
+/* Author:      Israel Fuller                                           */
+/* Creation:    September 29, 2025                                      */
+/* Description: Clustering library used to cluster and search data with */
+/*              cosine similarity and Levenshtein similarity (aka. edit */
+/*              distance). Used by the "clustering driver".             */
+/*              For more information on how to use this library, see    */
+/*              string-similarity.md in the centrallix-sysdoc folder.   */
+/************************************************************************/
+
+#include <stdlib.h>
+#include <stdbool.h>
+
+#ifdef CXLIB_INTERNAL
+#include "xarray.h"
+#else
+#include "cxlib/xarray.h"
+#endif
+
+/*** 2147483629 is the signed int max, and is also a prime number.
+ *** Using this value ensures that the longest run of 0s will not
+ *** cause an int underflow with the current encoding scheme.
+ *** 
+ *** Unfortunately, we can't use a number this large yet because
+ *** kmeans algorithm creates densely allocated centroids with
+ *** `CA_NUM_DIMS` dimensions, so a large number causes it to fail.
+ ***/
+#define CA_NUM_DIMS 251 //2147483629 /* aka. The vector table size. */
```

Comment:
Avoid using a // comment within a #define.  The result is compiler/environment specific, though it doesn't seem to be causing a problem for us using the gcc toolchain.